component/azstorage/datalake.go

```diff
@@ -324,8 +280,8 @@ func (dl *Datalake) DeleteFile(name string) (err error) {
 func (dl *Datalake) DeleteDirectory(name string) (err error) {
 	log.Trace("Datalake::DeleteDirectory : name %s", name)
 
-	directoryURL := dl.Filesystem.NewDirectoryURL(filepath.Join(dl.Config.prefixPath, name))
-	_, err = directoryURL.Delete(context.Background(), nil, true)
+	directoryClient := dl.Filesystem.NewDirectoryClient(filepath.Join(dl.Config.prefixPath, name))
+	_, err = directoryClient.Delete(context.Background(), &directory.DeleteOptions{}) // TODO:: track2: nil, true have to be passed
```

Comment:
Use nil instead of empty options.
True indicates recursive deletion of directory. This is set to true in track2 for directory deletion.